Wording Improvement in Section 4.3 Registration (#280)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

To register a Signed Statement, the Transparency Service performs the following steps:

> 1. **Client authentication:** A Client authenticates the Transparency Service before registering Signed Statements on behalf of one or more Issuers.

Reviewer: This is a MAY.  The client does not HAVE to.

Author: This is inconsistent with other text in the security consideration section where the assumption is made that TLS-based communication security is made.

If you use TLS then authentication of the TS to the client is mandatory, authentication of the client to the TS is optional.

Reviewer 2: This is a good point.

I think the intention here is that the server authenticates the client, not that the client authenticates the TS

Reviewer: Jon has made the change the other way in SCRAPI and thus this is out of sync.

Author: Any suggestion on how to improve the text based on the discussion? @OR13 @roywill

> 1. **Client authentication:** A Client authenticates the Transparency Service before registering Signed Statements on behalf of one or more Issuers.

Reviewer 3: ```suggestion
1. **Client authentication:** A Client authenticates with the Transparency Service before registering Signed Statements on behalf of one or more Issuers.
```

> Authentication and authorization is implementation-specific, and out of scope of the SCITT architecture.

Reviewer 4: ```suggestion
Authentication and authorization are implementation-specific and out of scope of the SCITT architecture.
```

> 1. **Issuer Verification:** The Transparency Service MUST validate the Issuer's identity Claims, which may be different than the Client identity.

Reviewer 5: 'syntactically' was a hard-fought compromise in a long discussion about how far down the stack of turtles an implementation should have to go in order to validate identity claims. For example, a signature should be well-formed and locally verify the signature against the protected public key, but whole chains, CRLs, OCSP etc checks do not need to be done (necessarily).

Author: If it means a lot to you then leave the syntactical term in there. I am just pointing out that the actual issuer verification is more than just a syntax check since it really requires cryptographic verification.

Reviewer 6: Constraining the verification requirement to depend only on information available directly in-line with the claims was the desired effect. That principal means a lot, the specific word 'syntactic' does not.

Author: @robinbryce Using only information from the COSE headers is fine. Like you, I would call this "syntactic"

Author: Any suggestion on what to do about the term 'syntactically' now? @robinbryce @JAG-UK

Reviewer 2: I think what is important is trusting the issuer's keys... the claims used to discover them are hints.

Reviewer 3: LGTM to latest changes

> 1. **Issuer Verification:** The Transparency Service MUST validate the Issuer's identity Claims, which may be different than the Client identity.

Reviewer 6: ```suggestion
1. **Issuer Verification:** The Transparency Service MUST syntactically verify the Issuer's identity Claims, which may be different than the Client identity. Only information local to the signed statement need be considered.
```

"validate the Issuer's identity Claims" remains ambiguous to me. I prefer that registration policies are the vehicle for ts's to express different, use case based, constraints on what is accepted on a ledger. If we must have a 'minimum' bar, verification of claims based exclusively on the local contents of the signed statement is what I support.

Reviewer 3: ```suggestion
1. **Issuer Verification:** The Transparency Service MUST perform signature verification, as defined in [RFC 9052 4.4](https://datatracker.ietf.org/doc/html/rfc9052#section-4.4), and MAY perform additional checks as part of its Registration Policy.
```

Reviewer 6: That is much better. THanks

> 1. **Issuer Verification:** The Transparency Service MUST perform signature verification, as defined in [RFC 9052 4.4](https://datatracker.ietf.org/doc/html/rfc9052#section-4.4), and MAY perform additional checks as part of its Registration Policy.

Reviewer 3: ```suggestion
1. **Issuer Verification:** The Transparency Service MUST perform signature verification, as defined in {{Section 4.4 of RFC9052}}, and MAY perform additional checks as part of its Registration Policy.
```

1. **Signature verification:** The Transparency Service MUST verify the signature of the Signed Statement, as described in {{RFC9360}}, using the signature algorithm and verification key of the Issuer.
1. **Signed Statement validation:** The Transparency Service MUST check that the Signed Statement includes the required protected headers.

> The Transparency Service MAY verify the Signed Statement payload.

Reviewer 2: ```suggestion
The Transparency Service MAY validate the Signed Statement payload, in order to enforce domain specific registration policies that apply to specific content types.
```

> The Transparency Service MAY validate the Signed Statement payload, in order to enforce domain specific registration policies that apply to specific content types.

Reviewer 4: ```suggestion
The Transparency Service MAY validate the Signed Statement payload in order to enforce domain specific registration policies that apply to specific content types.
```
Co-authored-by: Henk Birkholz <henkbirkholz@users.noreply.github.com>
Co-authored-by: Orie Steele <OR13@users.noreply.github.com>

> 1. **Signed Statement validation:** The Transparency Service MUST check that the Signed Statement includes the required protected headers.
> The Transparency Service MAY validate the Signed Statement payload in order to enforce domain specific registration policies that apply to specific content types.

Reviewer 3: Proposing removing line 555 as a SCITT Instance MAY do anything that's not explicitly called out as MUST NOT in the draft. 
@rjb4standards, this means your comment is still valid, without this sentence, and we can create more brevity.
```suggestion
1. **Signed Statement validation:** The Transparency Service MUST check that the Signed Statement includes the required protected headers.
```

> The Transparency Service MAY validate the Signed Statement payload in order to enforce domain specific registration policies that apply to specific content types.

Reviewer 7: DNS domain? Domain as in expertise?

Reviewer 8: BCG supports the proposed changes at 555

Reviewer 9: Apologies, some how my comment (earlier got lost)

Strictly speaking Line 555 is not necessary.

if at all one wishes to introduce this, I suggest it on the following lines.

Certain Transparency Services MAY choose to validate the Signed Statement payload for certain content types, based on their custom policies. However such validation is out of scope of this Architecture.

Reviewer 3: @yogeshbdeshpande, the PR refers to content, meaning the content of a content-type, for instance, the contenttype could be applicaiton/json, which isn't enough. The current PR leaves this more flexible, and is shorter.

Reviewer 3: @yogeshbdeshpande, on the editors call, we decided to move this PR forward. Please open a PR with some additional suggestions if needed.

Reviewer: Would rather keep the current proposal as less is more in this case.

1. **Signed Statement validation:** The Transparency Service MUST check that the Signed Statement includes the required protected headers listed above.
The Transparency Service MAY verify the Statement payload format, content and other optional properties.
1. **Signed Statement validation:** The Transparency Service MUST check that the Signed Statement includes the required protected headers.
The Transparency Service MAY validate the Signed Statement payload, in order to enforce domain specific registration policies that apply to specific content types.
1. **Apply Registration Policy:** The Transparency Service MUST check the attributes required by a Registration Policy are present in the protected headers.
  Custom Signed Statements are evaluated given the current Transparency Service state and the entire Envelope, and may use information contained in the attributes of named policies.
1. **Register the Signed Statement** to the Append-only Log.
1. **Return the Receipt**, which MAY be asynchronous from Registration.
The Transparency Service MUST be able to provide a Receipt for all registered Statements.

> A Receipt for a Signed Statement MAY be provided asynchronously.

Reviewer 3: This line is redundant with 559:
> 1. **Return the Receipt**, which MAY be asynchronous from Registration.
